packages/maskbook/src/plugins/RedPacket/types.ts

```diff
@@ -83,6 +106,31 @@ export interface RedPacketHistoryInMask {
     password: string
     shares: number
 }
+interface RedPacketCreator {
+    name: string
+    is_random: boolean
+    total: string
+    total_remaining: string
+    creation_time: number
+    last_updated_time: number
+    duration: number
+    chain_id: number
+    token: NativeTokenDetailed | ERC20TokenDetailed
+    creator: {
+        name: string
+        address: string
+    }
+    claimers: {
+        name: string
+        address: string
+    }[]
+    address: string
+}
+
+interface NFTRedPacketCreator extends Omit<RedPacketCreator, 'token' | 'is_random' | 'total_remaining'> {
```

Comment:
'total_remaining' should be preserved, since it can be calculated out by `bit_status`.

Reply:
> total_remaining could be preserved, since it can be calculated out by bit_status.

Then `total_remaining` becomes a computed-value. Here should be merely the raw data from `thegraph`.